src/evm/protocol/rfq/client.rs

```diff
@@ -0,0 +1,99 @@
+use std::pin::Pin;
+
+use async_trait::async_trait;
+use futures::{Stream, StreamExt};
+
+use crate::{
+    evm::protocol::rfq::{indicative_price::IndicativePrice, state::BindingQuote},
+    models::GetAmountOutParams,
+    protocol::errors::SimulationError,
+};
+
+#[async_trait]
+pub trait RFQClientSource: Send + Sync {
+    async fn next_price_update(&mut self)
+        -> Result<Vec<Box<dyn IndicativePrice>>, SimulationError>;
+
+    async fn request_binding_quote(
```

Comment:
Would be nice to have a small description explaining what you think are the responsibilities of each method